packages/platform-android/src/commands/runAndroid/runOnAllDevices.js

```diff
@@ -7,103 +7,85 @@
  * @flow
  */
 
-import {spawnSync, execFileSync} from 'child_process';
-import {logger} from '@react-native-community/cli-tools';
+import chalk from 'chalk';
+import {execFileSync} from 'child_process';
+import {logger, CLIError} from '@react-native-community/cli-tools';
 import adb from './adb';
 import tryRunAdbReverse from './tryRunAdbReverse';
 import tryLaunchAppOnDevice from './tryLaunchAppOnDevice';
+import type {FlagsT} from '.';
 
-function getCommand(appFolder, command) {
+function getTaskName(appFolder, command) {
   return appFolder ? `${appFolder}:${command}` : command;
 }
 
+function toPascalCase(value: string) {
+  return value[0].toUpperCase() + value.slice(1);
+}
+
 function runOnAllDevices(
-  args: Object,
+  args: FlagsT,
   cmd: string,
   packageNameWithSuffix: string,
   packageName: string,
   adbPath: string,
 ) {
   try {
-    const gradleArgs = [];
-
-    if (args.installDebug) {
-      gradleArgs.push(getCommand(args.appFolder, args.installDebug));
-    } else if (args.variant) {
-      gradleArgs.push(
-        `${getCommand(
-          args.appFolder,
-          'install',
-        )}${args.variant[0].toUpperCase()}${args.variant.slice(1)}`,
-      );
-    } else if (args.flavor) {
-      logger.warn('--flavor has been deprecated. Use --variant instead');
-      gradleArgs.push(
-        `${getCommand(
-          args.appFolder,
-          'install',
-        )}${args.flavor[0].toUpperCase()}${args.flavor.slice(1)}`,
-      );
-    } else {
-      gradleArgs.push(getCommand(args.appFolder, 'installDebug'));
-    }
+    const gradleArgs = [
+      args.task
+        ? getTaskName(args.appFolder, args.task)
+        : getTaskName(args.appFolder, 'install' + toPascalCase(args.variant)),
```

Comment:
idea: why don't we make args.task default value to be 'install + args.variant' and handle that where options are defined?

Either way, this was the piece I was looking for.

Reply:
Good call